flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```diff
@@ -322,77 +324,80 @@ public void visit(RemoteMethodCallActionNode remoteMethodCallActionNode) {
-                    if (optLocation.isEmpty()) {
-                        throw new IllegalStateException("Location not found for the variable symbol: " +
-                                variableSymbol);
-                    }
-                    NonTerminalNode parent = CommonUtil.findNode(variableSymbol, CommonUtils.getDocument(project,
-                            optLocation.get()).syntaxTree()).get().parent().parent();
-                    if (parent.kind() != SyntaxKind.MODULE_VAR_DECL) {
-                        throw new IllegalStateException("Parent is not a module variable declaration: " + parent);
-                    }
-                    ModuleVariableDeclarationNode moduleVariableDeclarationNode =
-                            (ModuleVariableDeclarationNode) parent;
-                    Optional<ExpressionNode> optInitializer = moduleVariableDeclarationNode.initializer();
-                    if (optInitializer.isEmpty()) {
-                        throw new IllegalStateException("Initializer not found for the module variable declaration: " +
-                                moduleVariableDeclarationNode);
+                FunctionDefinitionNode funcDef = (FunctionDefinitionNode) member;
+                if (funcDef.functionName().text().equals("init")) {
+                    for (StatementNode statement : ((FunctionBodyBlockNode) funcDef.functionBody()).statements()) {
+                        if (statement.kind() != SyntaxKind.ASSIGNMENT_STATEMENT) {
+                            continue;
+                        }
+                        AssignmentStatementNode assignmentStmtNode = (AssignmentStatementNode) statement;
+                        if (!assignmentStmtNode.varRef().toSourceCode().trim()
+                                .equals(expressionNode.toSourceCode())) {
+                            continue;
+                        }
+                        ImplicitNewExpressionNode newExpressionNode = getNewExpr(assignmentStmtNode.expression());
+                        genAgentData(newExpressionNode, classSymbol);
                     }
-                    ImplicitNewExpressionNode newExpressionNode = getNewExpr(optInitializer.get());
-                    genAgentData(newExpressionNode, classSymbol.get());
-                    break;
                 }
             }
         } else {
-            startNode(NodeKind.REMOTE_ACTION_CALL, expressionNode.parent());
-            setFunctionProperties(functionName, expressionNode, remoteMethodCallActionNode, functionSymbol,
-                    classSymbol.get().getName().orElseThrow());
+            for (Symbol moduleSymbol : semanticModel.moduleSymbols()) {
+                if (moduleSymbol.kind() != SymbolKind.VARIABLE) {
+                    continue;
+                }
+                VariableSymbol variableSymbol = (VariableSymbol) moduleSymbol;
+                if (!variableSymbol.getName().orElseThrow().equals(expressionNode.toSourceCode())) {
```

Comment:
Do we need to throw an exception for having `null` name?

Reply:
This was existing code. I just moved the code. Lets keep it as it is no avoid regression issues